kernel/yosys.h

```diff
@@ -155,6 +155,14 @@ extern Tcl_Obj *Tcl_ObjSetVar2(Tcl_Interp *interp, Tcl_Obj *part1Ptr, Tcl_Obj *p
 #  define YS_NORETURN
 #endif
 
+#if __cplusplus >= 201703L
+#  define YS_FALLTHROUGH [[fallthrough]];
+#elif defined(__GNUC__)
+#  define YS_FALLTHROUGH [[gnu::fallthrough]];
+#elif defined(__clang__)
+#  define YS_FALLTHROUGH [[clang::fallthrough]];
```

Comment:
This breaks on pre-C++17 MSVC.

Reply:
Doh, forgot to declare a fallback. Should it default to empty string or to a `/* fall through */` comment?

Reply:
As far as I can see the only compiler that will be affected by the fallback is MSVC, which doesn't have an implicit fallthrough warning anyway: https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warnings-by-compiler-version. Definining it as empty by default should be fine then.